dnd-troops-service/src/test/java/cz/muni/fi/pa165/w2018/dndtroops/service/facade/HeroFacadeImplTest.java

```diff
@@ -0,0 +1,162 @@
+package cz.muni.fi.pa165.w2018.dndtroops.service.facade;
+
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.HeroChangeDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.HeroDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.HeroSearchDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.RoleDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.backend.entity.Role;
+import cz.muni.fi.pa165.w2018.dndtroops.backend.enums.Race;
+import cz.muni.fi.pa165.w2018.dndtroops.api.facade.HeroFacade;
+import cz.muni.fi.pa165.w2018.dndtroops.backend.entity.Group;
+import cz.muni.fi.pa165.w2018.dndtroops.backend.entity.Hero;
+import cz.muni.fi.pa165.w2018.dndtroops.service.BeanMappingService;
+import cz.muni.fi.pa165.w2018.dndtroops.service.GroupService;
+import cz.muni.fi.pa165.w2018.dndtroops.service.HeroService;
+import cz.muni.fi.pa165.w2018.dndtroops.service.RoleService;
+import cz.muni.fi.pa165.w2018.dndtroops.service.configuration.ServiceConfiguration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.*;
+
+/**
+ * Tests fro HeroFacade implementation
+ *
+ * @author Marek Valko
+ */
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class HeroFacadeImplTest extends AbstractTestNGSpringContextTests {
+    @Mock
+    private HeroService heroService;
+    private GroupService groupService;
```

Comment:
GroupService should have @Mock annotation

Reply:
GroupService and Roleservice are not used, i will delete them, as well as unsued imports